sqa/fuel/writer/parquet.py

```diff
@@ -0,0 +1,576 @@
+from sqa.writer.parquet import TableBuilder, Column, BaseParquetWriter, add_index_column, add_size_column
+from .model import BlockHeader, Transaction, TransactionInput, TransactionOutput, Block, Receipt, \
+    Policies, TransactionInputContract, OutputContract
+
+
+def JSON():
+    return pyarrow.string()
+
+
+class BlockTable(TableBuilder):
+    def __init__(self):
+        self.number = Column(pyarrow.int32())
+        self.hash = Column(pyarrow.string())
+        self.da_height = Column(pyarrow.int32())
+        self.transactions_root = Column(pyarrow.string())
+        self.transactions_count = Column(pyarrow.uint64())
+        self.message_receipt_root = Column(pyarrow.string())
+        self.message_receipt_count = Column(pyarrow.uint64())
+        self.prev_root = Column(pyarrow.string())
+        self.application_hash = Column(pyarrow.string())
+        self.time = Column(pyarrow.uint64())
+
+    def append(self, block: BlockHeader) -> None:
+        self.number.append(block['height'])
+        self.hash.append(block['hash'])
+        self.da_height.append(int(block['daHeight']))
+        self.transactions_root.append(block['transactionsRoot'])
+        self.transactions_count.append(int(block['transactionsCount']))
+        self.message_receipt_root.append(block['messageReceiptRoot'])
+        self.message_receipt_count.append(int(block['messageReceiptCount']))
+        self.prev_root.append(block['prevRoot'])
+        self.application_hash.append(block['applicationHash'])
+        self.time.append(int(block['time']))
+
+
+class TransactionTable(TableBuilder):
+    def __init__(self):
+        self.block_number = Column(pyarrow.int32())
+        self.transaction_index = Column(pyarrow.int32())
+        self.hash = Column(pyarrow.string())
+        self.input_asset_ids = Column(pyarrow.list_(pyarrow.string()))
+        self.input_contracts = Column(pyarrow.list_(pyarrow.string()))
+        self.gas_price = Column(pyarrow.uint64())
+        self.script_gas_limit = Column(pyarrow.uint64())
+        self.maturity = Column(pyarrow.uint32())
+        self.mint_amount = Column(pyarrow.uint64())
+        self.mint_asset_id = Column(pyarrow.string())
+        self.tx_pointer = Column(pyarrow.string())
+        self.is_script = Column(pyarrow.bool_())
+        self.is_create = Column(pyarrow.bool_())
+        self.is_mint = Column(pyarrow.bool_())
+        self.type = Column(pyarrow.string())
+        self.witnesses = Column(pyarrow.list_(pyarrow.string()))
+        self.receipts_root = Column(pyarrow.string())
+        self.script = Column(pyarrow.string())
+        self.script_data = Column(pyarrow.string())
+        self.bytecode_witness_index = Column(pyarrow.int32())
+        self.bytecode_length = Column(pyarrow.uint64())
+        self.salt = Column(pyarrow.string())
+        self.storage_slots = Column(pyarrow.list_(pyarrow.string()))
+        self.raw_payload = Column(pyarrow.string())
+        self.policies = Column(pyarrow.struct([
```

Comment:
i made it flattened at first but then switched to this approach. i guess it's easier to request - request 1 fields instead of 4. and overhead on loaded data isn't big. also i had mentioned this `policies` field on our call i thought we concluded that it's okay to store and serve such fields without flattening